### ISSUE TITLE
Doesn’t work for <img>

### ISSUE THREAD

**Issue by pepelsbey** [Mrtenz/gatsby-remark-static-images#1]
As I noticed, `gatsby-remark-images` works fine with images embedded in Markdown using HTML. This is how MD is supposed to work: mixed with HTML. The typical use case would be a custom wrapper around your image or just a class name you need to add:

```html
<figure>
    <img src="images/1.png">
    <figcaption>Fig. #1 Image</figcaption>
</figure>
```

In case of `gatsby-remark-static-images` it just doesn’t work.

**Comment by Mrtenz** [comment]
Do you mean inside a JSX file or in a markdown file?

**Comment by pepelsbey** [comment]
In a markdown file, see [example file](https://raw.githubusercontent.com/web-standards-ru/new/master/content/articles/top-themes-and-frameworks/index.md) and the [project itself](https://github.com/web-standards-ru/new)

**Comment by Mrtenz** [comment]
Looks like this is a limitation in Remark itself. Remark doesn't parse the raw HTML nodes in the [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree). For example, your code would be parsed as:

```js
{
  type: 'html',
  value: '<figure>\n<img src="images/1.png" />\n<figcaption>Fig. #1 Image</figcaption>\n</figure>'
}
```

`gatsby-remark-images` [manually parses the HTML](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-images/src/index.js#L256) in order to solve this issue. I'll see if I can do something similar for this plugin.

**Comment by pepelsbey** [comment]
It would be wonderful if `gatsby-remark-static-images` could be a simple replacement for overcomplicated `gatsby-remark-images`. At least that’s what I was hoping for. Let me know if I can help!

**Comment by Mrtenz** [comment]
I just added some logic to parse images in HTML tags and changed the JavaScript to TypeScript. For example, the HTML from your example should get parsed as

```html
<figure>
  <img src="/static/1-[hash].png" />
  <figcaption>Fig. #1 Image</figcaption>
</figure>
```

I did some testing and everything seemed to work, but can you try it in your project and see if everything works as expected? It's not published to npm yet, so you can use it like this:

* Clone the repository
* Run `yarn install` (or `npm install`)
* Run `yarn link` (or `npm link`)
* In your own project, run `yarn link gatsby-remark-static-images` (or `npm link gatsby-remark-static-images`)

You should be able to use the local files then.

**Comment by pepelsbey** [comment]
Sorry, it took a while. One of our developers has tried it and still got relative `"images/1.jpg"` path. Also, the package itself relies on `yarn` though it’s not explicitly a dependency:

```sh
> gatsby-remark-static-images@1.0.0 prepublish /Users/pepelsbey/Desktop/gatsby-remark-static-images
> yarn run build

sh: yarn: command not found
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! gatsby-remark-static-images@1.0.0 prepublish: `yarn run build`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the gatsby-remark-static-images@1.0.0 prepublish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pepelsbey/.npm/_logs/2019-02-04T10_28_12_738Z-debug.log
```

**Comment by pepelsbey** [comment]
Tried it myself as well: installed yarn, linked globally, linked locally — the same result (

**Comment by FFxSquall** [comment]
`"prepublish": "yarn run build"` -> `"prepublish": "yarn build"`

**Comment by pepelsbey** [comment]
Right, just got a report that it worked. Maybe there’s something wrong with the linking method you proposed? Or maybe we just performed it wrong. I guess should be possible to install the package directly from GitHub.

**Comment by pepelsbey** [comment]
OK, it works :) Thanks! Could you please release it?

**Comment by Mrtenz** [comment]
Great, thank you for testing. Published it as [`gatsby-remark-static-images@1.1.0`](https://www.npmjs.com/package/gatsby-remark-static-images/v/1.1.0).

**Comment by pepelsbey** [comment]
Thank you so much! You saved us a lot of time :)
